Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ContentGraph.php

```diff
@@ -67,6 +70,10 @@ final public function getSubgraph(
     ): ContentSubgraphInterface {
         $index = $contentStreamId->value . '-' . $dimensionSpacePoint->hash . '-' . $visibilityConstraints->getHash();
         if (!isset($this->subgraphs[$index])) {
+            $privileges = $this->privilegeProvider->getPrivileges($visibilityConstraints);
+            if (!$privileges->isContentStreamAllowed($contentStreamId)) {
+                throw new \RuntimeException(sprintf('No access to content stream "%s"', $contentStreamId->value), 1681306937);
+            }
```

Comment:
does it make sense to check the privileges here already?

Reply:
I would say yes (if we can, like it is done here). Ofc people could work around this if they manually instanciated ContentSubgraph; but we could never prevent this.

Reply:
rename to sth like `$this->privilegeProvider->isFetchingOfNodesAllowedInSubgraph($contentStreamId, $dimensionSpacePoint)`